ksqldb-engine/src/test/java/io/confluent/ksql/integration/DependentStatementsIntegrationTest.java

```diff
@@ -0,0 +1,170 @@
+/*
+ * Copyright 2022 Confluent Inc.
+ *
+ * Licensed under the Confluent Community License (the "License"); you may not use
+ * this file except in compliance with the License.  You may obtain a copy of the
+ * License at
+ *
+ * http://www.confluent.io/confluent-community-license
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OF ANY KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+package io.confluent.ksql.integration;
+
+import static java.lang.String.format;
+
+import io.confluent.kafka.schemaregistry.client.SchemaRegistryClient;
+import io.confluent.ksql.util.KsqlConfig;
+import io.confluent.ksql.util.PersistentQueryMetadata;
+import io.confluent.ksql.util.QueryMetadata;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.TimeUnit;
+import java.util.stream.Collectors;
+import kafka.zookeeper.ZooKeeperClientException;
+import org.apache.kafka.test.IntegrationTest;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.ClassRule;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.experimental.categories.Category;
+import org.junit.rules.RuleChain;
+import org.junit.rules.Timeout;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+@Category({IntegrationTest.class})
+public class DependentStatementsIntegrationTest {
+
+  private static final Logger log = LoggerFactory.getLogger(DependentStatementsIntegrationTest.class);
+
+  private static final IntegrationTestHarness TEST_HARNESS = IntegrationTestHarness.build();
+
+  @ClassRule
+  public static final RuleChain CLUSTER_WITH_RETRY = RuleChain
+      .outerRule(Retry.of(3, ZooKeeperClientException.class, 3, TimeUnit.SECONDS))
+      .around(TEST_HARNESS);
+
+  public TestKsqlContext ksqlContext;
+
+  @Rule
+  public final Timeout timeout = Timeout.seconds(120);
+
+  private final List<QueryMetadata> toClose = new ArrayList<>();
+
+  @Before
+  public void before() throws Exception {
+    TEST_HARNESS.before();
+    ksqlContext  = TEST_HARNESS.ksqlContextBuilder()
+        .withAdditionalConfig(
+            KsqlConfig.SCHEMA_REGISTRY_URL_PROPERTY,
+            "http://foo:8080")
+        .build();
+
+    ksqlContext.before();
+
+    toClose.clear();
+  }
+
+  @After
+  public void after() {
+    toClose.forEach(QueryMetadata::close);
+    ksqlContext.after();
+    TEST_HARNESS.after();
+  }
+
+  @Test
+  public void shouldCreateDependentTopicWithDefaultReplicationInSandbox() {
```

Comment:
This test doesn't actually test a topic dependency, does it? I thought the dependency issue fixed in the previous PR was if the first CREATE STREAM creates a topic and then the second one uses the newly created topic. (Second statement should reference the topic, not the stream.)

This dependency case (CS followed by CSAS) should already be tested in other integration tests throughout the codebase.

Reply:
No, the fix was about "inherit the replication factor". The second statement creates an output topic, and was not able to create it, because it did neither inherit the replication factor from upstream topic, nor did if fetch default replication factor from Kafka.

Happy to remove this test again -- I thought you asked to add it. I can also add a different one that use two CS statements, but it also seems redundant?

Reply:
Oh, huh! I totally misunderstood the original ticket for that issue then, haha. If this integration test covers the fix, then that sounds great to me. Thanks for adding it!